collector/instance.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"database/sql"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver/v4"
+)
+
+const (
+	MYSQL   = "mysql"
+	MARIADB = "mariadb"
```

Comment:
Let's go with the upstream naming.

```suggestion
	MySQL   = "mysql"
	MariaDB = "mariadb"
```